pkg/apis/networking/validation/validation_test.go

```diff
@@ -1566,267 +1566,181 @@ func TestValidateIngressUpdate(t *testing.T) {
 	}
 }
 
+type netIngressTweak func(ingressClass *networking.IngressClass)
+
+func makeValidIngressClassUpdateCustom(name, controller string, tweaks ...netIngressTweak) *networking.IngressClass {
+	ingressClass := &networking.IngressClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: networking.IngressClassSpec{
+			Controller: controller,
+		},
+	}
+
+	for _, fn := range tweaks {
+		fn(ingressClass)
+	}
+	return ingressClass
+}
+
+func makeIngressClassSpecParamsCustom(apiGroup *string, kind, name string, scope, namespace *string) *networking.IngressClassParametersReference {
+	ingressClassParametersReference := &networking.IngressClassParametersReference{}
+
+	if apiGroup != nil {
+		ingressClassParametersReference.APIGroup = apiGroup
+	}
+
+	if kind != "" {
```

Comment:
You don't need to test these - setting them to "" is fine (it's just a noop)

Reply:
@thockin 
Got it! now I am returning the type

```
func makeIngressClassParams(apiGroup *string, kind, name string, scope, namespace *string) *networking.IngressClassParametersReference {
	return &networking.IngressClassParametersReference{
		APIGroup:  apiGroup,
		Kind:      kind,
		Name:      name,
		Scope:     scope,
		Namespace: namespace,
	}
}
```